tests/unit/bash/bash_pkg_conditional_rpm.bats.jinja

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+
+# Mock the "rpm" command
+# mocked package: coconut-4.5.17-8.fc31
+# The mock is used to avoid the need to have a specific RPM package in a
+# specific version installed on the system where this test is run.
+function rpm()
+{
+    if [[ "$*" == "--quiet -q coconut" ]] ; then
+        return 0
+    elif [[ "$*" ==  "-q --queryformat %{VERSION}-%{RELEASE} coconut" ]] ; then
+        echo "4.5.17-8.fc31"
+        return 0
+    else
+        echo "BATS mock for rpm doesn't support this command"
+        return 1
+    fi
+}
+export -f rpm
+
+@test "bash_pkg_conditional_rpm - test package presence" {
+    {{{ bash_pkg_conditional_rpm("coconut") | indent(4) }}}
+}
+
+@test "bash_pkg_conditional_rpm - test package version" {
+    {{{ bash_pkg_conditional_rpm("coconut", "5", "<") | indent(4) }}}
+    {{{ bash_pkg_conditional_rpm("coconut", "5.0", "<") | indent(4) }}}
+    {{{ bash_pkg_conditional_rpm("coconut", "5.0.0-1", "<") | indent(4) }}}
+    {{{ bash_pkg_conditional_rpm("coconut", "5.0.0-1.fc31", "<") | indent(4) }}}
+
+    {{{ bash_pkg_conditional_rpm("coconut", "5", "<=") | indent(4) }}}
+    {{{ bash_pkg_conditional_rpm("coconut", "5.0", "<=") | indent(4) }}}
+    {{{ bash_pkg_conditional_rpm("coconut", "5.0.0-1", "<=") | indent(4) }}}
+    {{{ bash_pkg_conditional_rpm("coconut", "5.0.0-1.fc31", "<=") | indent(4) }}}
+
+    {{{ bash_pkg_conditional_rpm("coconut", "4.5.17-9.fc31", "<") | indent(4) }}}
+    {{{ bash_pkg_conditional_rpm("coconut", "4.5.17-10.fc31", "<") | indent(4) }}}
+    {{{ bash_pkg_conditional_rpm("coconut", "4.5.18-1.fc31", "<") | indent(4) }}}
+    {{{ bash_pkg_conditional_rpm("coconut", "4.5.18-12.fc31", "<") | indent(4) }}}
+
+    {{{ bash_pkg_conditional_rpm("coconut", "4.5.17-9.fc31", "<=") | indent(4) }}}
+    {{{ bash_pkg_conditional_rpm("coconut", "4.5.17-10.fc31", "<=") | indent(4) }}}
+    {{{ bash_pkg_conditional_rpm("coconut", "4.5.18-1.fc31", "<=") | indent(4) }}}
+    {{{ bash_pkg_conditional_rpm("coconut", "4.5.18-12.fc31", "<=") | indent(4) }}}
+
+    {{{ bash_pkg_conditional_rpm("coconut", "4.5.17-7.fc31", ">") | indent(4) }}}
+    {{{ bash_pkg_conditional_rpm("coconut", "4.5.16-1.fc31", ">") | indent(4) }}}
+    {{{ bash_pkg_conditional_rpm("coconut", "4.1.1-9.fc31", ">") | indent(4) }}}
+    {{{ bash_pkg_conditional_rpm("coconut", "4.1.1-1.fc31", ">") | indent(4) }}}
+
+    {{{ bash_pkg_conditional_rpm("coconut", "4.5.17-7.fc31", ">=") | indent(4) }}}
+    {{{ bash_pkg_conditional_rpm("coconut", "4.5.16-1.fc31", ">=") | indent(4) }}}
+    {{{ bash_pkg_conditional_rpm("coconut", "4.1.1-9.fc31", ">=") | indent(4) }}}
+    {{{ bash_pkg_conditional_rpm("coconut", "4.1.1-1.fc31", ">=") | indent(4) }}}
+
+    {{{ bash_pkg_conditional_rpm("coconut", "4.5.17-8.fc31", "<=") | indent(4) }}}
```

Comment:
I expect the usage `coconut>=4.5` to be the most common - if you have an upstream version that is known to have a feature, then it is the upper bound of all version numbers that may have it, as a feature may be backported to an eariler version.
Therefore, various distributions could specify their distribution-specific version specifiers if they are after it, and a catch-all spec would look like a safe comparison against the upstream version, not considering the release number and maybe even not the minor number.

Anyway, I would add such example here, the good thing is that it works.